lib/routes/shared/security_pin/backup_phrase/verify_backup_phrase_page.dart

```diff
@@ -0,0 +1,199 @@
+        padding: EdgeInsets.only(left: 16.0, right: 16.0, bottom: 40.0, top: 24.0),
+        child: new Column(
+            mainAxisSize: MainAxisSize.max, crossAxisAlignment: CrossAxisAlignment.start, children: _buildVerificationFormContent()),
+      ),
+    );
+  }
+
+  Padding _buildInstructions() {
+    return Padding(
+      padding: EdgeInsets.only(left: 72, right: 72),
+      child: Text(
+        "Please type words number ${_randomlySelectedIndexes[0] + 1}, ${_randomlySelectedIndexes[1] + 1} and ${_randomlySelectedIndexes[2] + 1} of the generated backup phrase.",
+        style: theme.backupPhraseInformationTextStyle.copyWith(color: theme.BreezColors.white[300]),
+        textAlign: TextAlign.center,
+      ),
+    );
+  }
+
+  List<Widget> _buildVerificationFormContent() {
+    List<Widget> selectedWordList = List.generate(
+      _randomlySelectedIndexes.length,
+      (index) {
+        return Padding(
+          padding: const EdgeInsets.only(bottom: 16),
+          child: TextFormField(
+            decoration: new InputDecoration(
+              labelText: "${_randomlySelectedIndexes[index] + 1}",
+            ),
+            style: theme.FieldTextStyle.textStyle,
+            validator: (text) {
+              if (text.length == 0 || text.toLowerCase().trim() != _mnemonicsList[_randomlySelectedIndexes[index]]) {
+                setState(() {
+                  _hasError = true;
+                });
+              }
+              return null;
+            },
+          ),
+        );
+      },
+    );
+    if (_hasError)
+      selectedWordList
+        ..add(Text(
+          "Failed to verify words. Please write down the words and try again.",
+          style: theme.errorStyle,
+        ));
+    return selectedWordList;
+  }
+
+  Future _createBackupPhrase(SecurityModel securityModel, UserProfileBloc userProfileBloc, BackupBloc backupBloc) async {
+    var createBackupPhraseAction = CreateBackupPhrase(widget._mnemonics);
+    userProfileBloc.userActionsSink.add(createBackupPhraseAction);
+    createBackupPhraseAction.future.then((_) {
+      _updateSecurityModel(securityModel, securityModel.copyWith(backupKeyType: BackupKeyType.PHRASE), userProfileBloc, backupBloc);
+    }).catchError((err) {
+      promptError(context, "Internal Error", Text(err.toString(), style: theme.alertStyle,));
+    });
+  }
+
+  _selectIndexes() {
+    List list = List.generate(23, (i) => i);
+    list.shuffle();
+    for (var i = 0; i < 3; i++) {
+      _randomlySelectedIndexes.add(list[i]);
+    }
+    _randomlySelectedIndexes.sort();
+  }
+
+  Future _updateSecurityModel(
+      SecurityModel oldModel, SecurityModel newModel, UserProfileBloc userProfileBloc, BackupBloc backupBloc) async {
+    var action = UpdateSecurityModel(newModel);
+    userProfileBloc.userActionsSink.add(action);
+    Navigator.popUntil(context, ModalRoute.withName("/security"));
+    action.future.then((_) {
+      backupBloc.backupNowSink.add(true);
```

Comment:
The following logic somehow looks duplicate as it exist in the parent page `SecurityPage`.
Can we just pop here and return the security model to the parent page and from there use the regular flow of updating the model, backup, etc... ?